pkg/appliance/functions.go

```diff
@@ -901,6 +914,25 @@ func downloadDockerImageBundle(args imageBundleArgs) {
 		headers["Authorization"] = "Basic " + base64.StdEncoding.EncodeToString([]byte(auth))
 	}
 
+	requestRetry := func(client *http.Client, req *http.Request) (*http.Response, error) {
+		var err error
+		var res *http.Response
+		err = backoff.Retry(func() error {
+			res, err = client.Do(req)
+			if err != nil {
+				return backoff.Permanent(err)
+			}
+			if res.StatusCode != http.StatusOK {
+				return fmt.Errorf("Recieved %s status", res.Status)
+			}
+			return nil
+		}, backoff.NewExponentialBackOff())
+		if err != nil {
+			return nil, err
+		}
+		return res, nil
+	}
```

Comment:
I'm not quite sure who this library works but shouldn't we set a max retry?

Reply:
Library doesn't really have a max retry, but a default timeout (which we can control, of course) of 15 minutes, which I think is fine.